src/veins/base/toolbox/Signal.h

```diff
@@ -24,227 +24,602 @@
-
-    friend Signal operator*(const Signal& lhs, const Signal& rhs);
-    friend Signal operator*(const Signal& lhs, double rhs);
-    friend Signal operator*(double lhs, const Signal& rhs);
-
-    friend Signal operator/(const Signal& lhs, const Signal& rhs);
-    friend Signal operator/(const Signal& lhs, double rhs);
-    friend Signal operator/(double lhs, const Signal& rhs);
-
-    Signal operator<<(uint16_t n);
-    Signal operator>>(uint16_t n);
-
-    friend inline simtime_t calculateStart(const Signal& lhs, const Signal& rhs);
-    friend inline simtime_t calculateDuration(const Signal& lhs, const Signal& rhs);
-
-    uint64_t getBitrate() const;
-    void setBitrate(uint64_t rate);
 
     /**
-     * Returns a pointer to the arrival module. It returns NULL if the signal
-     * has not been sent/received yet, or if the module was deleted
-     * in the meantime.
+     * Divide the power levels by a constant.
+     *
+     * @param value power level to divide by in milliwatt
      */
-    cModule* getReceptionModule() const;
+    Signal& operator/=(const double value);
+    ///@}
 
     /**
-     * Returns pointers to the gate from which the signal was sent and
-     * on which gate it arrived. A NULL pointer is returned
-     * for new (unsent) signal.
+     * Convert a signal to a human readable representation.
+     *
+     * @param os the outputstream to write to
+     * @param s the signal to convert
      */
-    cGate* getReceptionGate() const;
+    friend std::ostream& operator<<(std::ostream& os, const Signal& s);
 
     /**
-     * Returns a pointer to the sender module. It returns NULL if the signal
-     * has not been sent/received yet, or if the sender module got deleted
-     * in the meantime.
+     * Calculate the common start of two signals, i.e. their minimum start times.
+     *
+     * @param lhs the first signal
+     * @param rhs the second signal
      */
-    cModule* getSendingModule() const;
+    friend inline simtime_t calculateStart(const Signal& lhs, const Signal& rhs);
 
     /**
-     * Returns pointers to the gate from which the signal was sent and
-     * on which gate it arrived. A NULL pointer is returned
-     * for new (unsent) signal.
+     * Calculate the combined duration of two signals.
+     *
+     * @param lhs the first signal
+     * @param rhs the second signal
      */
-    cGate* getSendingGate() const;
-
-    /** @brief Saves the arrival sender module information form message. */
-    void setReceptionSenderInfo(const cMessage* const pMsg);
-
-    void addAttenuation(uint16_t freqIndex, double factor);
-    void addUniformAttenuation(double factor);
+    friend inline simtime_t calculateDuration(const Signal& lhs, const Signal& rhs);
 
 private:
-    void includeAbsoluteIndex(size_t freqIndex);
+    double getMinInRange(size_t freqIndexLow, size_t freqIndexHigh) const;
+    double getMaxInRange(size_t freqIndexLow, size_t freqIndexHigh) const;
 
     Spectrum spectrum;
```

Comment:
```suggestion
     Spectrum spectrum = nullptr;
```

Reply:
Spectrum is a value type. Therefore it must be default initialized here.